components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -1199,9 +1199,14 @@ public static String getSPTenantDomainFromClientId(String clientId) {
         try {
             OAuthAppDO oAuthAppDO = OAuth2Util.getAppInformationByClientId(clientId);
             return OAuth2Util.getTenantDomainOfOauthApp(oAuthAppDO);
-        } catch (IdentityOAuth2Exception | InvalidOAuthClientException e) {
+        } catch (IdentityOAuth2Exception e) {
             log.error("Error while getting oauth app for client Id: " + clientId, e);
```

Comment:
If we are doing a fallback here, shouldn't this be a WARN or a DEBUG log?

Reply:
We are separating client and server errors. **IdentityOAuth2Exception**  is considered to be server error and we also catch **InvalidOAuthClientException** for which we will have debug logs.